astropy/utils/tests/test_data.py

```diff
@@ -1131,20 +1146,13 @@ def osraiser(dirnm, linkto, pkgname=None):
         ),
     ],
 )
-def test_read_unicode(filename):
-    contents = get_pkg_data_contents(os.path.join("data", filename), encoding="utf-8")
-    assert isinstance(contents, str)
-    contents = contents.splitlines()[1]
-    assert contents == "האסטרונומי פייתון"
-
-    contents = get_pkg_data_contents(os.path.join("data", filename), encoding="binary")
-    assert isinstance(contents, bytes)
-    x = contents.splitlines()[1]
-    expected = (
-        b"\xff\xd7\x94\xd7\x90\xd7\xa1\xd7\x98\xd7\xa8\xd7\x95\xd7\xa0\xd7\x95"
-        b"\xd7\x9e\xd7\x99 \xd7\xa4\xd7\x99\xd7\x99\xd7\xaa\xd7\x95\xd7\x9f"[1:]
-    )
-    assert x == expected
+def test_read_unicode(filename, encoding, expected_type, expected_lines):
+    contents = get_pkg_data_contents(os.path.join("data", filename), encoding=encoding)
+    assert type(contents) is expected_type
+    # using str.splitlines here for portability:
+    # newlines are decoded as different character sequences
+    # on Unix ('\n') and Windows ('\r\n')
+    assert contents.splitlines() == expected_lines
```

Comment:
Eero reminded me of this. Probably also have to make sure this is not a bug in `get_pkg_data_contents`. I will investigate when I get the chance.

Reply:
So, is the debate `splitlines()` isn't  necessary? I find that it is still needed on Windows. In fact, the old code had it too so I don't understand what the controversy is. Have to step away. More later.